File: benchmarks/competitors/bench_loop.js
// ==============================================================================
// 🟩 NODE.JS BENCHMARK: 10,000,000 ARITHMETIC LOOP ITERATIONS (benchmarks/bench_loop.js)
// ==============================================================================

const start = process.hrtime.bigint();
let total = 0;
for (let i = 1; i <= 10000000; i++) {
    total += i % 7;
}
const elapsedMs = Number(process.hrtime.bigint() - start) / 1e6;

console.log("Node.js Loop 10,000,000 Iterations Complete!");
console.log(`Sum: ${total}`);
console.log(`Elapsed Time: ${elapsedMs.toFixed(2)} ms`);

File: benchmarks/competitors/bench_loop.zig
// ==============================================================================
// ⚡ ZIG BENCHMARK: 10,000,000 ARITHMETIC LOOP ITERATIONS (benchmarks/bench_loop.zig)
// ==============================================================================

const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();

    const start = std.time.nanoTimestamp();

    var total: u64 = 0;
    var i: u64 = 1;
    while (i <= 10_000_000) : (i += 1) {
        total += (i % 7);
    }

    const end = std.time.nanoTimestamp();
    const elapsed_ns = end - start;
    const elapsed_ms = @as(f64, @floatFromInt(elapsed_ns)) / 1_000_000.0;

    try stdout.print("Zig Loop 10,000,000 Iterations Complete!\n", .{});
    try stdout.print("Sum: {d}\n", .{total});
    try stdout.print("Elapsed Time: {d:.2} ms\n", .{elapsed_ms});
}
